Automattic/jetpack
# Hooks: Ensure translation loading is not triggered too early.

Continuation of https://github.com/Automattic/jetpack/pull/39696

## Proposed changes:

* This PR ensures that both the Tiled Gallery and Contact Info blocks are loaded later to ensure translation loading is not triggered too early, resulting in loading issues and error log warnings (and Contact Info block file has been condensed into one file instead of two)
* It also ensures `init_sync_cron_jobs` is hooked into `init`, instead of `plugins_loaded`, for the same reason.
* Also, Jetpack's `load_modules` is hooked into `after_setup_theme'` to ensure it also runs after translations have been loaded.
* A function has been created to load VideoPress' page suffix as that calls a translation function and runs too early otherwise, so the function itself is now hooked into `admin_menu`.
  * Due a test issue with a previous commit where this was pulled out of the VP init function, the admin menu tests have also been updated based on reviews as well (technically unrelated to this PR)

Notice that is being prevented with these changes:

```
PHP Notice:  Function _load_textdomain_just_in_time was called <strong>incorrectly</strong>. Translation loading for the <code>jetpack</code> domain was triggered too early. This is usually an indicator for some code in the plugin or theme running too early. Translations should be loaded at the <code>init</code> action or later. Please see <a href="https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.7.0.) in /var/www/html/wp-includes/functions.php on line 6099
```

### Other information:


## Jetpack product discussion

See links in https://github.com/Automattic/jetpack/pull/39696

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Note - you may need to set a non-English (United States) language for your site (and install the corresponding translations, which is done via `/wp-admin/update-core.php` - I found I had to navigate through the site a bit before coming back to the core updates page when translations would now be available - updating translations seems to be needed rather than just confirming they are up to date.). 

You should also tail your sites error logs to check for the above notice, which in testing with trunk and WP 6.7 should be visible.

**Self-hosted testing steps:**

To replicate the issue: 

* On a free site, connected to WordPress.com,  install the WordPress Beta Tester plugin
* Go to Tools > Beta Testing, and set the plugin to use Beta/RC Only (as we're now in the RC stage of the 6.7 release - nightly will give you 6.8)
* Go to Dashboard > Updates, and update to the most recent version of WP.
* You likely will need to update translations at this stage.
* You should be seeing errors in your log related to `_load_textdomain_just_in_time`. 

To test the fix:
* Apply this PR
* Go to Posts > Add New
* Ensure the Tiled Gallery and Contact Info blocks work as expected.
* Ensure sync actions still work. Try schedule a sync checksum via the Jetpack debugger, for example, or add a new post and ensure the change is reflected on WPcom (`https://wordpress.com/` `posts/mysite.com`).
* Also, test some modules to ensure they are loading correctly, particularly comments.
* Lastly, test adding and activating the VideoPress standalone plugin (if testing locally, ensure the PR is applied using the Beta tester plugin for the VideoPress plugin), and make sure no errors show up. The VideoPress 'Jetpack VideoPress' menu item name should display without issue.
* None of the testing should trigger the notice shared under 'Proposed changes' above.

As an aside, cleanup in one testing file can also be tested, by navigating to `projects/plugins/jetpack` if testing locally and running `jetpack docker phpunit -- --filter=test_jetpack_admin_menu_order` (though the tests will also pass in this PR).
